src/internal/middleware/context/interceptor.go

```diff
@@ -0,0 +1,99 @@
+package context
```

Comment:
This is the name of a standard library package.  Almost every package we have references `context` so this package will always be difficult to refer to.  Maybe something like `ctxmiddleware` or `ctxintercept`